### DISCUSSION TITLE
LlamaCode compatible?

### DISCUSSION THREAD

**Discussion by wiiiktor** [guidance-ai/guidance#484] [Q&A] [answered]
Is LlamaCode also compatible with Guidance, with the most powerful control features that require endpoint integration? 

If the answer is "yes, but it depends", can you point me to an exact LlamaCode model on HuggingFace that I can use?

**Comment by marcotcr** [comment, marked as answer]
It's compatible. I've spent the past few days playing with [this](https://huggingface.co/Phind/Phind-CodeLlama-34B-v2) (although I'm using `llama.cpp`), and it works fine with constraints and everything.

**Reply by marcotcr** [reply]
[Here](https://github.com/guidance-ai/guidance/blob/main/notebooks/code_generation.ipynb) is an example
